k210.svd

```diff
@@ -998,11 +998,223 @@
             <groupName>FFT</groupName>
             <baseAddress>0x42000000</baseAddress>
             <registers>
-                <!-- TODO -->
                 <register>
-                    <name>dummy</name>
-                    <description>Dummy register: this peripheral is not implemented yet</description>
+                    <name>input_fifo</name>
+                    <description>FFT input data fifo</description>
                     <addressOffset>0x00</addressOffset>
+                    <size>64</size>
+                    <fields>
+                        <field>
+                            <name>input_fifo</name>
+                            <description>Input fifo</description>
+                            <bitRange>[63:0]</bitRange>
+                        </field>
+                    </fields>
+                </register>
+                <register>
+                    <name>ctrl</name>
+                    <description>FFT control register</description>
+                    <addressOffset>0x08</addressOffset>
+                    <size>64</size>
+                    <fields>
+                        <field>
+                            <name>point</name>
+                            <description>FFT calculation data length</description>
+                            <bitRange>[2:0]</bitRange>
+                            <enumeratedValues>
+                                <enumeratedValue>
+                                    <name>p512</name>
+                                    <description>512 point</description>
+                                    <value>0</value>
+                                </enumeratedValue>
+                                <enumeratedValue>
+                                    <name>p256</name>
+                                    <description>256 point</description>
+                                    <value>1</value>
+                                </enumeratedValue>
+                                <enumeratedValue>
+                                    <name>p128</name>
+                                    <description>128 point</description>
+                                    <value>2</value>
+                                </enumeratedValue>
+                                <enumeratedValue>
+                                    <name>p64</name>
+                                    <description>64 point</description>
+                                    <value>3</value>
+                                </enumeratedValue>
+                            </enumeratedValues>
+                        </field>
+                        <field>
+                            <name>mode</name>
+                            <description>FFT mode</description>
+                            <bitRange>[3:3]</bitRange>
+                            <enumeratedValues>
+                                <enumeratedValue>
+                                    <name>backward</name>
```

Comment:
According to [this](https://www.gnu.org/software/gsl/doc/html/fft.html), *forward FFT* is `FFT` and *backward FFT* is a variant of *inverse FFT*, `IFFT`. This is not consistent with the information from `fft.h` and `fft.c`.
I suggest naming these variants `fft` and `ifft`.
```suggestion
                                    <name>fft</name>
```

Reply:
I was really confused here too, so I used the naming from the enums that are used in the  example code, even though the comment seems to conflict. If you're sure about ifft/fft that's better naming, though.